Content.Server/ReagentOnItem/EntitySystems/SpaceGlueOnItemSystem.cs

```diff
@@ -0,0 +1,105 @@
+using Content.Shared.Hands;
+using Content.Shared.Interaction.Components;
+using Content.Shared.Popups;
+using Content.Shared.ReagentOnItem;
+using Robust.Shared.Timing;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Examine;
+using Content.Shared.IdentityManagement;
+
+namespace Content.Server.ReagentOnItem;
+
+public sealed class SpaceGlueOnItemSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly InventorySystem _inventory = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SpaceGlueOnItemComponent, GotEquippedHandEvent>(OnHandPickUp);
+        SubscribeLocalEvent<SpaceGlueOnItemComponent, ExaminedEvent>(OnExamine);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<SpaceGlueOnItemComponent, UnremoveableComponent, MetaDataComponent>();
+        while (query.MoveNext(out var uid, out var glue, out var _, out var meta))
+        {
+            if (_timing.CurTime < glue.TimeOfNextCheck)
+                continue;
+
+            // If false then the item should be droppable and the glue has worn off.
+            // Otherwise, it will just add more time and we will check again later!
+            if (!SetNextNextCanDropCheck(glue, uid))
+            {
+                RemCompDeferred<UnremoveableComponent>(uid);
+                RemCompDeferred<SpaceGlueOnItemComponent>(uid);
+            }
+        }
+    }
+
+    private void OnHandPickUp(Entity<SpaceGlueOnItemComponent> entity, ref GotEquippedHandEvent args)
+    {
+        // Check to see if the thing picking them up has non stick gloves.
+        // If they do, just let em pick the item up!
+        _inventory.TryGetSlotEntity(args.User, "gloves", out var gloves);
+        if (HasComp<NonStickSurfaceComponent>(gloves))
+            return;
+
+        // This means that the item will be stuck to their hand.
+        if (SetNextNextCanDropCheck(entity, entity))
+        {
+            _popup.PopupEntity(Loc.GetString("space-glue-on-item-hand-stuck", ("target", Identity.Entity(entity, EntityManager))), args.User, args.User, PopupType.MediumCaution);
+        }
+
+    }
+
+    /// <summary>
+    ///     This function sets the next time we will check in Update to see if the item
+    ///     is droppable or if there is still more reagent left on the item.
+    /// </summary>
+    /// <returns> Will return true if the item should still be stuck and false if the item should be droppable. </returns>
```

Comment:
This comment can stay.

`This function` can be removed since it's redundant.
Need to add that it checks whether there is no reagent left.